cmd/manager/main.go

```diff
@@ -210,3 +219,30 @@ func ensureDefaults(cfg *rest.Config, scheme *kruntime.Scheme) error {
 
 	return err
 }
+
+// generateMetrics creates a service, service monitor, starts a metrics server
+// and records the metrics to expose at the specified endpoint.
+
+func generateMetrics(cfg *rest.Config, namespace string) {
+
+	// Create a Service object to expose the metrics port.
+	service, err := metrics.ExposeMetricsPort(context.TODO(), customMetrics.MetricsPort)
+	if err != nil {
+		log.Errorf("Unable to expose metrics port: %v", err)
+	}
+
+	// Ensure that a nil pointer is not added to the slice.
+	services := []*v1.Service{}
+	if service != nil {
+		services = append(services, service)
+	}
+
+	// Create a serviceMonitor for the namespace and register the service created above to it.
+	_, err = metrics.CreateServiceMonitors(cfg, namespace, services)
+	if err != nil {
+		log.Errorf("Error Creating Service Monitor: %v", err)
```

Comment:
Same comment as above with respect to error handling. Is just logging appropriate?